### DISCUSSION TITLE
Object Graph and delete (another 4.xx question)

### DISCUSSION THREAD

**Discussion by Chicagoan2016** [MarimerLLC/csla#4240] [Questions]
Hi,
We have an object graph with child collection and grad-child collections.
When we delete the graph using deferred delete, we set the child collection to new collections by calling DataPortal.Create<>.
However, our root object still maintains its property values.
It's not a big issue since we clear out the UI but I wanted to know if there is a way to 'reset' our root object.
I remember we followed the approach from Project tracker.
We could move to Immediate deletion.

Kind Regards

**Comment by StefanOssendorf** [comment]
Hi.. Which version are you using? And as far as I know a rep place with create does not delete the (grand) children.. You have to clear the list and save the root. With that the portal logic on the server can go over the deleted items and delete them. 
Hth

**Reply by Chicagoan2016** [reply]
It's an old project that uses Csla 4.xx
